src/ziggurat/kafka_consumer/consumer.clj

```diff
@@ -1,28 +1,41 @@
 (ns ziggurat.kafka-consumer.consumer
   (:require [clojure.tools.logging :as log]
-            [ziggurat.kafka-consumer.consumer-handler :refer :all])
+            [ziggurat.kafka-consumer.consumer-handler :refer :all]
+            [ziggurat.util.map :as umap])
   (:import (java.util Map Properties)
            (org.apache.kafka.clients.consumer KafkaConsumer ConsumerConfig)
            (java.util.regex Pattern)))
 
+(def default-consumer-config
+  {:commit-interval-ms 15000
+   :max-poll-records 500
+   :session-timeout-ms-config 60000
+   :key-deserializer-class-config "org.apache.kafka.common.serialization.ByteArrayDeserializer"
+   :value-deserializer-class-config "org.apache.kafka.common.serialization.ByteArrayDeserializer"})
+
 (defn- build-consumer-properties-map
-  [consumer-group-config]
+  [{:keys [bootstrap-servers
+           consumer-group-id
+           max-poll-records
+           session-timeout-ms-config
+           commit-interval-ms
+           key-deserializer-class-config
+           value-deserializer-class-config]}]
   (doto (Properties.)
-    (.putAll {ConsumerConfig/BOOTSTRAP_SERVERS_CONFIG        (:bootstrap-servers consumer-group-config)
-              ConsumerConfig/GROUP_ID_CONFIG                 (:consumer-group-id consumer-group-config)
-              ConsumerConfig/MAX_POLL_RECORDS_CONFIG         (int (or (:max-poll-records  consumer-group-config) 500))
-              ConsumerConfig/SESSION_TIMEOUT_MS_CONFIG       (int (or (:session-timeout-ms-config  consumer-group-config) 60000))
-              ConsumerConfig/ENABLE_AUTO_COMMIT_CONFIG       false
-              ConsumerConfig/KEY_DESERIALIZER_CLASS_CONFIG   (or (:key-deserializer-class-config consumer-group-config)
-                                                                 "org.apache.kafka.common.serialization.ByteArrayDeserializer")
-              ConsumerConfig/VALUE_DESERIALIZER_CLASS_CONFIG (or (:value-deserializer-class-config consumer-group-config)
-                                                                 "org.apache.kafka.common.serialization.ByteArrayDeserializer")})))
-
+    (.putAll {ConsumerConfig/BOOTSTRAP_SERVERS_CONFIG        bootstrap-servers
+              ConsumerConfig/GROUP_ID_CONFIG                 consumer-group-id
+              ConsumerConfig/MAX_POLL_RECORDS_CONFIG         (int max-poll-records)
+              ConsumerConfig/SESSION_TIMEOUT_MS_CONFIG       (int session-timeout-ms-config)
+              ConsumerConfig/ENABLE_AUTO_COMMIT_CONFIG       true
```

Comment:
moving this to default config map can improve readability further.

Reply:
default-config-map ?

Reply:
done